orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/Message.kt

```diff
@@ -24,12 +24,50 @@ import com.netflix.spinnaker.orca.Task
 import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import java.util.*
 
 /**
  * Messages used internally by the queueing system.
  */
+
+@JsonTypeInfo(use = MINIMAL_CLASS, include = PROPERTY, property = "@class")
+interface Attribute {
+}
+
+data class MaxAttemptsAttribute(val maxAttempts: Int = -1) : Attribute {
+}
+
+data class AttemptsAttribute(var attempts: Int = 0) : Attribute {
+  fun increment() {
+    this.attempts = attempts + 1
+  }
+}
```

Comment:
I'm not clear why we don't just have 

    data class Attributes(
      val attempts: Int = 0,
      val maxAttempts: Int = -1
    )

which would avoid all the potential null issues elsewhere.